streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -211,28 +207,44 @@ public synchronized void start() {
      * and then wait for them to join.
      *
      * @throws IllegalStateException if process has not started yet
+     * @param timeout   how long to wait for {@link StreamThread}s to shutdown
+     * @param timeUnit  unit of time used for timeout
      */
-    public synchronized void close() {
+    public synchronized void close(final long timeout, final TimeUnit timeUnit) {
```

Comment:
I've added a no-arg `close` that defaults to 30 seconds. I've also left the other one. My personal take on this is that doing it via config is clumsy. Close methods should almost always have timeout parameters